include/neighbor/Runtime.h

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018-2019, Michael P. Howard.
+// This file is released under the Modified BSD License.
+
+// Maintainer: mphoward
+
+#ifndef NEIGHBOR_RUNTIME_H_
+#define NEIGHBOR_RUNTIME_H_
+
+#include <functional>
+#include <utility>
+
+#include <cuda_runtime.h>
+#ifdef __CUDACC__
+#define GPUCC
+#endif
+
+namespace neighbor
+{
+namespace gpu
+{
+
+/* errors */
+typedef cudaError_t error_t;
+enum error
+    {
+    success=cudaSuccess
+    };
+//! Coerce comparison of gpu::error with the native error as int
+inline bool operator==(error a, error_t b)
+    {
+    return (static_cast<int>(a) == static_cast<int>(b));
+    }
+//! Coerce comparison of gpu::error with the native error as int
+inline bool operator==(error_t a, error b)
+    {
+    return (static_cast<int>(a) == static_cast<int>(b));
+    }
+
+/* streams */
+typedef cudaStream_t stream_t;
+//! Create a GPU stream
+inline error_t streamCreate(stream_t* stream)
+    {
+    return cudaStreamCreate(stream);
+    }
+//! Synchronize a GPU stream
+inline error_t streamSynchronize(stream_t stream)
+    {
+    return cudaStreamSynchronize(stream);
+    }
+//! Destroy a GPU stream
+inline error_t streamDestroy(stream_t stream)
+    {
+    return cudaStreamDestroy(stream);
+    }
+
+/* memory */
+static const int memAttachGlobal = cudaMemAttachGlobal;
```

Comment:
In CUDA, these are defines. Do you think it's OK to make them variables (which gives them namespace scope), or should they also be defines?